fin/src/include/tensor.hpp

```diff
@@ -57,6 +57,31 @@ using status_t = cl_int;
 using status_t = int;
 #endif
 
+inline miopenTensorLayout_t GetMemLayout(const std::string& s)
+{
+
+    if(s == "NCHW")
```

Comment:
This should be a switch.

Reply:
iirc switches only work for integer values. This string would need to be recast as an enum.